flit_core/flit_core/wheel.py

```diff
@@ -138,8 +139,8 @@ def _add_file(self, full_path, rel_path):
         self.records.append((rel_path, hash_digest, size))
 
     @contextlib.contextmanager
-    def _write_to_zip(self, rel_path, mode=0o644):
-        sio = StringIO()
+    def _write_to_zip(self, rel_path, mode=0o644, newline='\n'):
+        sio = StringIO(newline=newline)
```

Comment:
I think passing `newline` around is not needed here; I see the csv module docs say to use `newline=''`, but `'\n'` (the default for StringIO) is equivalent:

> When writing output to the stream... If newline is '' or '\n', no translation takes place.

Reply:
Oh yeah, that's a good point. Reverted this part.